File: library/src/main/java/me/proxer/library/entity/ucp/UcpTopTenEntry.java
package me.proxer.library.entity.ucp;

import com.squareup.moshi.Json;
import lombok.Getter;
import lombok.Value;
import me.proxer.library.entity.ProxerIdItem;
import me.proxer.library.entity.info.Entry;
import me.proxer.library.enums.Category;
import me.proxer.library.enums.Medium;

/**
 * Entity representing a single entry in the history list.
 *
 * @author Ruben Gees
 */
@Value
public class UcpTopTenEntry implements ProxerIdItem {

    /**
     * {@inheritDoc}
     */
    @Getter(onMethod = @__({@Override}))
    @Json(name = "fid")
    private String id;

    /**
     * Returns the id of the associated {@link Entry}.
     */
    @Json(name = "eid")
    private String entryId;

    /**
     * Returns the name.
     */
    @Json(name = "name")
    private String name;

    /**
     * Returns the medium.
     */
    @Json(name = "medium")
    private Medium medium;

    /**
     * Returns the category.
     */
    @Json(name = "kat")
    private Category category;
}

File: library/src/main/kotlin/me/proxer/library/entity/ucp/UcpTopTenEntry.kt
package me.proxer.library.entity.ucp

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import me.proxer.library.entity.ProxerIdItem
import me.proxer.library.entity.info.Entry
import me.proxer.library.enums.Category
import me.proxer.library.enums.Medium

/**
 * Entity representing a single entry in the history list.
 *
 * @property entryId The id of the associated [Entry].
 * @property name The name.
 * @property medium The medium.
 * @property category The category.
 *
 * @author Ruben Gees
 */
@JsonClass(generateAdapter = true)
data class UcpTopTenEntry(
    @Json(name = "fid") override val id: String,
    @Json(name = "eid") val entryId: String,
    @Json(name = "name") val name: String,
    @Json(name = "medium") val medium: Medium,
    @Json(name = "kat") val category: Category
) : ProxerIdItem
